### PR TITLE
Refactoring the code and removing unwanted variables

### PR DESCRIPTION
**Pull request by somindatommy**
## Proposed changes in this pull request

Removes the channel related variable since they have defined in a central location. The PR also resolves a logical error self-registration flow and refactored the code.

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dto/ResendConfirmationDTO.java [hunk 1]
```diff
@@ -42,6 +42,11 @@
      */
     private String successCode;
 
+    /**
+     * Code to used to confirm password recovery when the notifications are managed externally.
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dto/ResendConfirmationDTO.java]: Shall we change the comment as "Code which is used to ...".

Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dto/ResendConfirmationDTO.java]: Fixed be669d5091be5c073ab01d1b036541dc18b37269

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java [hunk 2]
```diff
@@ -261,20 +260,22 @@ private boolean isNotificationsInternallyManaged(String tenantDomain, Boolean no
...
-                                        IdentityRecoveryConstants.VERIFIED_USER_PROPERTY_KEY);
-                        log.debug(message);
-                    }
-                }
+        if (MapUtils.isEmpty(propertyMap)) {
+            return false;
+        }
+        String verified = propertyMap.get(IdentityRecoveryConstants.VERIFIED_USER_PROPERTY_KEY);
+        if (StringUtils.isBlank(verified)) {
+            return false;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: let's add a debug log here

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java [hunk 3]
```diff
@@ -261,20 +260,22 @@ private boolean isNotificationsInternallyManaged(String tenantDomain, Boolean no
...
-                } catch (NumberFormatException e) {
-                    if (log.isDebugEnabled()) {
-                        String message = String
-                                .format("Value : %s given for property : %s is not a boolean. Therefore, "
-                                                + "returning false for the property.", verified,
-                                        IdentityRecoveryConstants.VERIFIED_USER_PROPERTY_KEY);
-                        log.debug(message);
-                    }
-                }
+        if (MapUtils.isEmpty(propertyMap)) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: let's add a debug log here

Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed: 5049039d9746db87efff623e4e1eba7f4f69fca3

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java (hunk 2 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed: 5049039d9746db87efff623e4e1eba7f4f69fca3

Comment (somindatommy) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java]: Fixed: 5049039d9746db87efff623e4e1eba7f4f69fca3

**Pull request merged**